app/controllers/password_controller.rb

```diff
@@ -60,10 +59,10 @@ def user_code
 
   def process_code
     if session[:email].nil? && params[:password_recovery_form][:email].nil?
-      flash.now[:errors] = t('email_missing')
-      render :forgot_form, status: :bad_request
+      flash.now[:errors] = t('password.email_missing')
+      redirect_to forgot_password_path
     else
-      @form = PasswordRecoveryForm.new(params[:password_recovery_form])
+      @form = PasswordRecoveryForm.new(params[:password_recovery_form] || {})
```

Comment:
I don't think we need the `|| {}` due to `PasswordRecoveryForm` setting a default value in the initialiser: `def initialize(params = {})`

Reply:
this wasn't because of the model, but when the POST request doesn't have `params` ... just tested this, and it fails earlier on line 61 anyway. I'll revert this line back...

Reply:
https://github.com/alphagov/verify-self-service/pull/213/commits/0aebccd395b84e091524316b3696cba2b01b2dbc